### PR TITLE
Visual studio fix

### PR DESCRIPTION
**Pull request by WolfgangVogl**
Removed two unused exception variabes (every compiler should have complained about it...).
    A (ugly) workaround for visual studio.... this prevents the C2888 Compiler error for line 1201 and 1216.
    Removed some boilerplate code for the using.

### CODE REVIEW

File: include/belle.hh [hunk 1]
```diff
@@ -1196,10 +1196,15 @@ private:
 
   private:
 
+#ifndef _MSC_VER
```
Comment (octobanana) [inline include/belle.hh:1199]: Since the 'if' has an 'else', instead of testing for false, could it test for true and swap the bodies around:
```cpp
#ifdef _MSC_VER
...
#else
...
#endif 
```

File: include/belle.hh [hunk 2]
```diff
@@ -1196,10 +1196,15 @@ private:
...
     // generic lambda for sending different types of responses
-    static auto const constexpr send = [](auto self, auto&& res) -> void
+    static auto constexpr send = [](auto self, auto&& res) -> void
+#else
+    template <typename SELF_T, typename RES_T>
```
Comment (octobanana) [inline include/belle.hh:1203]: To follow the naming convention used in other templates, could the variables be named:
```cpp
template <typename Self, typename Res>
```

File: include/belle.hh [hunk 3]
```diff
@@ -1196,10 +1196,15 @@ private:
...
+    static constexpr void send(SELF_T self, RES_T&& res)
+#endif
     {
-      using item_type = typename std::remove_reference<decltype(res)>::type;
+      using item_type = std::remove_reference_t<decltype(res)>;
```
Comment (octobanana) [inline include/belle.hh:1207]: This is nice, less boilerplate. Since it's unrelated to the Visual Studio fix, would it be better separated into its own commit or pull request?

File: include/belle.hh [hunk 4]
```diff
@@ -1348,7 +1348,7 @@ private:
             {
               return static_cast<int>(e);
             }
-            catch (std::exception const& e)
+            catch (std::exception const&)
```
Comment (octobanana) [inline include/belle.hh:1356]: In the future, I'd like to add a logging interface which will end up using this variable. To suppress the warning in the mean time, the following could be used:
```cpp
catch (std::exception const& e)
{
  boost::ignore_unused(e);

  return 500;
}
```

**Review by octobanana — CHANGES_REQUESTED** [review]

File: include/belle.hh (hunk 1 above)
Comment (WolfgangVogl) [reply include/belle.hh:1199]: I'd liked the idea of having the real/origin code above the workaround. But if you prefer it the other way that's no problem.

File: include/belle.hh (hunk 3 above)
Comment (WolfgangVogl) [reply include/belle.hh:1207]: Sorry, I'm not so firm with the github pull request mechanism. Realized to late that I need a new branch for every request.

File: include/belle.hh (hunk 4 above)
Comment (WolfgangVogl) [reply include/belle.hh:1356]: I dont see why you prefer adding boost::ignore_unused (to remove it later, then you add logging interface) over just adding the " e" part then you add the logging interface?!

**Comment by WolfgangVogl** [comment]
Created another pull request - sorry had to fork again to get it clear...
